### DISCUSSION TITLE
Deploy RKE2 nodes in a cluster with multiple CIDRs/subnets

### DISCUSSION THREAD

**Discussion by athulg93** [rancher/rke2#6396] [Q&A] [answered]
Hey,
so one of the requirements we have is to deploy RKE2 on VMs. The VMs will be on different ESXI hosts and that means they could be on different subnets.
e.g: 
vm 1 & 2: 172.18.46.2 | 172.18.46.3
vm 3 & 4: 192.18.11.2 | 192.18.11.30
vm 5 & 6: 172.46.152.11 | 172.46.152.40
When subnets change, so will the CIDR.

So does anyone know how we can setup RKE2 to deploy over multiple CIDrs/subnets?

**Comment by brandond** [comment, marked as answer]
RKE2 doesn't care if your nodes are on different subnets, so long as they can all reach each other. Or are you trying to have different nodes us different cluster-cidr and service-cidr values? Those values need to be set cluster-wide, and are not in any way related to the networks that your nodes are on.
